### DISCUSSION TITLE
DEC RX01/RX02 Format

### DISCUSSION THREAD

**Discussion by pmut** [keirf/greaseweazle#256] [Ideas]
Hi all,

I work with a v4 version to Image my vintage floppies quite successfully.
So many thanks and greetings to the project team :-) !

I have also a bigger amount of DEC stuff laying around where I would like to Image Floppy Disks too.
By now, as I think, there is no support for the RX01 or RX02 format available, so I wonder if it could be included... .

I would be available for testing. Native hardware is available for this ... .

Many Greetings
Ulrich

**Comment by keirf** [comment]
Not integrated into greaseweazle, but there is rx02 support in one of my other projects https://github.com/keirf/disk-utilities

**Comment by keirf** [comment]
And if that works out and you have some good raw images to share, I can work on integration into greaseweazle

**Reply by pmut** [reply]
OK, will have a look into.

To be sure to do the right things and that my understanding is right:
a.) Take a RAW Image with greaseweazle from RX01 (Tracks 77	Sectors 26 Heads 1	Sector Size 128 Encoding FM Capacity 256k)
b.) Take a RAW Image with greaseweazle from RX02 (Tracks 77	Sectors 26 Heads 1	Sector Size 256 Encoding FM Capacity 512k)
b.) disk_analyze of both images
c.) provide you access to the RAW images

**Reply by keirf** [reply]
Yes something like that. The command line for disk-analyse will look something like `disk-analyse --format=DEC-RX02 my.scp my.img` and obvious equivalent for dec-rx01.

**Comment by wohali** [comment]
Hi Keir, I can supply some images read this way, if I remember on the weekend.

**Comment by mkatzNU** [comment]
How can I get the greaseweazle to put a low level dec.rx01 or dec.rx02 low level format on a disk?  Do I need to take a formatted disk and read it and then write it back to the second disk?

**Reply by keirf** [reply]
You can write an empty (literally zero sized) IMG file and it will automatically get zero extended by gw and write zero into all sectors of your rx01 or rx02 disk. This may not be helpful though as the disk will not contain any filesystem metadata. Often a system's format tool will also write an empty filesystem. In this case you would want a formatted empty image file to write.

**Reply by keirf** [reply]
So if you can, best is to dump an existing disk to IMG file and then use that file to create new empty formatted disks.

**Reply by mkatzNU** [reply]
1.  If I understand you correctly. An image file of all zeros will create the appropriate sector headers without any data on them?
2.  Does erase do that or does erase just write zeros to the entire track?
3.  It would be really cool if there was a format command that used the format described in the diskdefs.cfg file.

Thanks.

**Reply by keirf** [reply]
1. Yes
2. Completely empty unformatted track with no sector headers
3. I suppose so. This would be equivalent to (1). All sectors defined, data contents all zero. No filesystem metadata.

**Reply by mkatzNU** [reply]
#1 or #3 is exactly what I am looking for.

Though I have created an image disk from the RX02 with an OS/8 directory on it from my PDP-8/E.

Thank you.

I have already successfully copied an RX02 disk (single density headers with double density data) with two SA-800's.

I would like to make 2 suggestions.  Would you like me to post them in the actions section?

1.  Add a format command that is the equivalent to number 1 above.
2.  Add a copy command that would copy a disk
  A.  From one drive to another without saving an image to disk
  B.  From one drive to another with saving an image to disk.
  C.  With a single drive, reading data into memory and disk (optional) and then prompting for the other disk to write the image too.

I don't think either of these would require a firmware change.

Thanks again.

**Comment by daydream58** [comment]
I found this 8" in a briefcase under my bed.  I know I took it out of a DEC system in early 1980 and it's been in that briefcase ever since!  Does anyone have the ability to actually read the data on it?

![dec 1979](https://github.com/user-attachments/assets/29885417-7952-4f1d-94b6-fb9455b693ba)
![dec disk](https://github.com/user-attachments/assets/0d891e25-9e85-4fd7-8620-88da27bd1cfe)
